application/F3DStarter.cxx

```diff
@@ -164,8 +165,16 @@ int F3DStarter::Start(int argc, char** argv)
   this->Internals->Parser.GetOptions(
     this->Internals->AppOptions, this->Internals->DynamicOptions, files);
 
-  // Set verbosity level early from command line
-  F3DInternals::SetVerboseLevel(this->Internals->AppOptions.VerboseLevel);
+  const bool renderToStdout = this->Internals->AppOptions.Output == "-";
+  if (renderToStdout)
+  {
+    f3d::log::setVerboseLevel(f3d::log::VerboseLevel::ERROR);
+  }
+  else
+  {
+    // Set verbosity level early from command line
+    F3DInternals::SetVerboseLevel(this->Internals->AppOptions.VerboseLevel);
+  }
```

Comment:
```suggestion
  std::streambuf* originalStdout = std::cout.rdbuf();
  if (renderToStdout)
  {
    std::cout.rdbuf(std::cerr.rdbuf());
  }

  // Set verbosity level early from command line
  F3DInternals::SetVerboseLevel(this->Internals->AppOptions.VerboseLevel);
```

Reply:
interesting suggestion, how do other cli software do it ?

Reply:
Honestly I think we shouldn't even do anything about the verbose level automatically. Let's let the user specify the right verbosity level manually?
About external libraries potentially writing to std::cout, it's not our problem and we can still provide patches upstream if needed?

Reply:
And the default verbosity behavior is only writing in std::cerr anyway?

Reply:
> Honestly I think we shouldn't even do anything about the verbose level automatically.

That part definitely,  let's not change it magically.

> Let's let the user specify the right verbosity level manually?
> About external libraries potentially writing to std::cout, it's not our problem and we can still provide patches upstream if needed?

These 2 I guess can be argued both ways, we could do nothing and let the users find out if their output get corrupted, then either tell them they did it wrong in the first case, or go patch and rebuild in the second case.
Redirecting would be the "safe" way but I can't say if it's a good/nice/clean thing to do, or if anybody else does it that way as @mwestphal asked...

> And the default verbosity behavior is only writing in std::cerr anyway?

I haven't checked so I'll trust you on that, in which case it's fine as long as we trust libraries (looking at you OCCT...) to not randomly print directly.

Reply:
Redirect to stderr should be easy. We just have to override that: vtkOutputWindow::GetDisplayStream
So let's decide. I'm not against the idea.

Reply:
that sounds like a good idea to me.